ElectronClient/app/gui/NotebookPropertiesDialog.tsx

```diff
@@ -0,0 +1,66 @@
+import * as React from 'react';
+import { useState, useEffect } from 'react';
+
+const { _ } = require('lib/locale.js');
+const { themeStyle } = require('../theme.js');
+const DialogButtonRow = require('./DialogButtonRow.min');
+
+const Folder = require('lib/models/Folder.js');
+
+interface NotebookPropertiesDialogProps {
+	theme: number,
+	folderId: string,
+	onClose: Function,
+}
+
+export default function NotebookPropertiesDialog(props: NotebookPropertiesDialogProps) {
+	console.info('Render NotebookPropertiesDialog');
+
+	const [folder, formFolder] = useState<any>({});
+
+	const theme = themeStyle(props.theme);
+
+	useEffect(() => {
+		async function fetchFolder() {
+			if (!props.folderId) {
+				formFolder({folder: null});
```

Comment:
Better to throw an exception as it won't work with an empty folder.

Reply:
Can definitely do.